content/docs/internals/architecture.md

```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
 
-- Authorization Service
+At a high level, when the Authentication service first receives a request from the Proxy service, it:
+1. Sends an HTTP request on behalf of the client to the IdP
+1. Completes the authentication flow with the IdP and extracts relevant session data, such as device credentials, user ID, issuance and expiration times, OAuth tokens, and any OIDC claims (scopes) provided by the IdP
+1. Saves session data to a local session cookie and redirects the client with session data encrypted in URL parameters to the Proxy service
+1. Verifies user identity after the session expires
 
-  - Processes policy to determine permissions for each service
-  - Handles authorization check for all user sessions
-  - Directs Proxy service to initiate Authentication flow as required
-  - Provides additional security related headers for upstream services to consume
+### Authorization service
+The Authorization service processes policies to determine what permissions the client has. Each request sent by the client must first go through an authorization check before the Proxy service proxies the request. 
 
-- Databroker Service
+When the Authorization service receives a request from the Proxy service, the following actions take place:
+1. The Authorization service first looks for a session cookie, which contains the client’s JWT. If no session cookie is present, the Authorization service sends a request to the Authentication service to establish session identity. 
+1. Once a session cookie is in place, the Proxy service redirects the request to the Authorization service so it can determine permissions based on JWT claims and policy. 
+1. The Authorization service then constructs security headers based on JWT claims, which the Proxy service forwards to upstream applications.
+1. With each subsequent request, the Authorization service checks the session expiration time and communicates session changes to the Databroker. 
 
-  - Retrieves identity provider related data such as group membership
-  - Stores and refreshes identity provider access and refresh tokens
-  - Provides streaming authoritative session and identity data to Authorize service
-  - Stores session and identity data in persistent storage
+### Databroker service
+The Databroker service persists session and identity-related data and streams session state updates to other services. It also functions as an identity manager in that it's responsible for refreshing user sessions against the IdP until a Pomerium session has expired. 
+
+The points below outline the Databroker’s role in the request and session lifecycle:
+
+- Once a client is authenticated, the Proxy server redirects the session data and identity information to the Databroker. The Databroker persists the data. 
+- The Databroker streams authoritative session and identity data to the Authorization service, keeping the two services in sync.
```

Comment:
The Databroker no longer streams authoritative session and identity data to the Authorization service. That data is queried on-demand by the Authorization service.